covenant/covenant.go

```diff
@@ -436,30 +440,70 @@ func (ce *CovenantEmulator) delegationsToBatches(dels []*types.Delegation) [][]*
 	return batches
 }
 
-func RemoveAlreadySigned(localKey *btcec.PublicKey, dels []*types.Delegation) []*types.Delegation {
+func RemoveNotInCommittee(paramCache ParamsGetter, covenantSerializedPk []byte, dels []*types.Delegation) []*types.Delegation {
 	sanitized := make([]*types.Delegation, 0, len(dels))
-	localKeyBytes := schnorr.SerializePubKey(localKey)
 
 	for _, del := range dels {
-		delCopy := del
```

Comment:
Any reason we had a copy here?

Reply:
We create it due to how golang pre 1.22 handle loop variables - https://go.dev/blog/loopvar-preview

Given that this is fixed in golang 1.22 and covenant emulator requires at least 1.23. We can probably get rid of it though it would be good to have some test that pointers to structs are properly added to new slice.

Reply:
maybe I did not understood, but with delCopy https://github.com/babylonlabs-io/covenant-emulator/pull/91/commits/a3c8f15ac3821800d464455d96eb8bdc080240d4

or without `delCopy` https://github.com/babylonlabs-io/covenant-emulator/pull/91/commits/d0e3cba93eee4ad8363798592a14b6d073c1510a

It fails if a pointer is modified :thinking:

Reply:
so it is not really about pointer modification but about what `sanitized := make([]*types.Delegation, 0, len(dels))` will contain after whole sanitization loop.

In old go versions (before v1.22), failing to make the `delCopy` and using `sanitized = append(sanitized, del)` to append to collection, would result in collection that  contains pointer to one element which is last element `dels []*types.Delegation` (so you would have n pointers to the same element)

So the test whether this behaviour does not exist is to:
- have initial collection with `n` delegations where every delegation is different
- run `SanitizeDelegations` function that should filter few of them
- check the returned collection contain different elements

Reply:
aahhh, understood! Will do that, thanks

Reply:
done at https://github.com/babylonlabs-io/covenant-emulator/pull/91/commits/e99764a7541be76291f837d24653075489f9a17b